MyMovies/Model Controllers/MovieController.swift

```diff
@@ -7,52 +7,215 @@
 //
 
 import Foundation
+import CoreData
+
+enum HTTPMethod: String {
+    case get = "GET"
+    case put = "PUT"
+    case post = "POST"
+    case delete = "DELETE"
+}
 
 class MovieController {
     
-    private let apiKey = "4cc920dab8b729a619647ccc4d191d5e"
-    private let baseURL = URL(string: "https://api.themoviedb.org/3/search/movie")!
-    
-    func searchForMovie(with searchTerm: String, completion: @escaping (Error?) -> Void) {
+    var searchedMovies: [MovieRepresentation] = []
         
-        var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        private let fireBase = URL(string: "https://mymovies2-8d1a3.firebaseio.com/")!
+        private let apiKey = "4cc920dab8b729a619647ccc4d191d5e"
+        private let baseURL = URL(string: "https://api.themoviedb.org/3/search/movie")!
         
-        let queryParameters = ["query": searchTerm,
-                               "api_key": apiKey]
+        init() {
+            fetchMoviesFromServer()
+        }
         
-        components?.queryItems = queryParameters.map({URLQueryItem(name: $0.key, value: $0.value)})
+        func createMovie(with title: String, identifier: UUID, hasWatched: Bool) {
+            let movie = Movie(title: title, hasWatched: hasWatched, identifier: identifier)
+            put(movie: movie)
+            CoreDataStack.shared.save()
+        }
         
-        guard let requestURL = components?.url else {
-            completion(NSError())
-            return
+        func update(movie: Movie, title: String, hasWatched: Bool) {
+            
+            movie.title = title
+            movie.hasWatched = hasWatched
+            
+            put(movie: movie)
+            
+            CoreDataStack.shared.save()
+        }
```

Comment:
Great function call and set up. Only thing here is to stay consistent with how you are building your functions. In your createMovie function you have no whitespaces. In this one you have three. 
I recommend getting rid of the whitespaces to keep your code clean and consistent.